ONSdigital/dp-cantabular-csv-exporter
# Added component tests and fixed some issues with running multiple com…

…ponent tests

### What

- Added extra component tests for:
  - public file generation (already existed, but has been improved to support `healthcheck` subscription)
  - private file generation
  - check no consumption on unhealthy state

- Improved drainTopic to drain any topic, and use it to drain both topics used in tests. This is needed for tests that do not consume messages, e.g. 'not consume on unhealthy'

- Fixed race condition between endpoints creation and service running by starting the service in a step after all dependencies have been defined.

### How to review

- Make sure changes make sense
- Make sure new component tests make sense
- Make sure component tests consistently pass

### Who can review

anyone
